File: backend/api/DTO/DeckDto.cs
using api.Models;

namespace api.DTO;

public class DeckDto
{
    public int Id { get; set; }
    public string Name { get; set; }
    public int UserId { get; set; }

    public DeckDto()
    {
    }

    public DeckDto(Deck deck)
    {
        Id = deck.Id;
        Name = deck.Name;
        UserId = deck.UserId;
    }
}

File: desktop/lib/dto/deck_dto.dart
class DeckDto {
  String name;
  int userId;

  DeckDto({
    required this.name,
    required this.userId,
  });

  factory DeckDto.fromJson(Map<String, dynamic> json) => DeckDto(
        name: json["name"],
        userId: json["userId"],
      );

  Map<String, dynamic> toJson() => {
        "name": name,
        "userId": userId,
      };
}
